src/coordinated_workers/coordinator.py

```diff
@@ -766,43 +772,27 @@ def s3_ready(self) -> bool:
         except S3NotFoundError:
             return False
 
-    @property
-    def peer_addresses(self) -> List[str]:
-        """If a peer relation is present, return the addresses of the peers."""
+    def _get_peer_data(self, field: str) -> Dict[ops.model.Unit, str]:
+        """Return a mapping of unit -> <unit databag field value> for all units in the peer relation (excluding self)."""
         peers = self._peers
-        relation = self.model.get_relation("peers")
-        # get unit addresses for all the other units from a databag
-        addresses = []
-        if peers and relation:
-            addresses = [relation.data.get(unit, {}).get("local-ip") for unit in peers]
-            addresses = list(filter(None, addresses))
+        relation = self.model.get_relation(self._peer_relation)
 
-        # add own address
-        if self._local_ip:
-            addresses.append(self._local_ip)
+        if not (peers and relation):
+            return {}
 
-        return addresses
+        result = {
+            unit: relation.data.get(unit, {}).get(field)
+            for unit in peers
+            if relation.data.get(unit, {}).get(field) is not None
+        }
+        return cast(Dict[ops.model.Unit, str], result)
```

Comment:
I think this can be simplified:
we only have _peers if we have a peer relation. If one of them is None, then both are. Also the typing of the first early-exit return statement can be improved by initializing a typed dict and returning that instead

```suggestion
        peer_relation = self.model.get_relation(self._peer_relation)
        out: Dict[ops.model.Unit, str] = {}
        
        if not peer_relation:
            return out

        out.update({
            unit: (value := peer_relation.data.get(unit, {}).get(field))
            for unit in peer_relation.units  # or self._units; they're equivalent
            if value is not None
        })
        
        return out
```